Build failure on some time64-mitigated architectures:

Hi, I am trying to rebuild powerman v2.4.3 on Debian Sid/Unstable.

Please check https://buildd.debian.org/status/logs.php?pkg=powerman&ver=2.4.3-1%7Eexp1 , which shows package build failure on some architectures, including the important ones like riscv64 and armhf.

Looking into the build logs such as https://buildd.debian.org/status/fetch.php?pkg=powerman&arch=armhf&ver=2.4.3-1%7Eexp1&stamp=1724457836&raw=0 , the following build failure is shown:

```
Making all in powerman
make[3]: Entering directory '/<<PKGBUILDDIR>>/src/powerman'
gcc -DHAVE_CONFIG_H -I. -I../../config  -I../../src/liblsd -I../../src/libcommon -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -c -o powerman.o powerman.c
/bin/bash ../../libtool  --tag=CC   --mode=link gcc -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security  -flto=auto -ffat-lto-objects -Wl,-z,relro -Wl,-z,now  -o powerman powerman.o ../../src/liblsd/liblsd.la ../../src/libcommon/libcommon.la -lgenders 
libtool: link: gcc -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -flto=auto -ffat-lto-objects -Wl,-z -Wl,relro -Wl,-z -Wl,now -o powerman powerman.o  ../../src/liblsd/.libs/liblsd.a ../../src/libcommon/.libs/libcommon.a -lgenders
rm -f pm
ln -s powerman pm
gcc -DHAVE_CONFIG_H -I. -I../../config  -I../../src/liblsd -I../../src/libcommon -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -c -o arglist.o arglist.c
gcc -DHAVE_CONFIG_H -I. -I../../config  -I../../src/liblsd -I../../src/libcommon -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -c -o client.o client.c
gcc -DHAVE_CONFIG_H -I. -I../../config  -I../../src/liblsd -I../../src/libcommon -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -c -o debug.o debug.c
gcc -DHAVE_CONFIG_H -I. -I../../config  -I../../src/liblsd -I../../src/libcommon -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Werror -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -c -o device.o device.c
device.c: In function ‘_process_delay’:
device.c:1335:56: error: format ‘%ld’ expects argument of type ‘long int’, but argument 4 has type ‘__time64_t’ {aka ‘long long int’} [-Werror=format=]
 1335 |             act->vpf_fun(act->client_id, "delay(%s): %ld.%-6.6ld", dev->name,
      |                                                      ~~^
      |                                                        |
      |                                                        long int
      |                                                      %lld
 1336 |                     delay.tv_sec, delay.tv_usec);
      |                     ~~~~~~~~~~~~                        
      |                          |
      |                          __time64_t {aka long long int}
device.c:1335:64: error: format ‘%ld’ expects argument of type ‘long int’, but argument 5 has type ‘__suseconds64_t’ {aka ‘long long int’} [-Werror=format=]
 1335 |             act->vpf_fun(act->client_id, "delay(%s): %ld.%-6.6ld", dev->name,
      |                                                          ~~~~~~^
      |                                                                |
      |                                                                long int
      |                                                          %-6.6lld
 1336 |                     delay.tv_sec, delay.tv_usec);
      |                                   ~~~~~~~~~~~~~                 
      |                                        |
      |                                        __suseconds64_t {aka long long int}
cc1: all warnings being treated as errors
make[3]: *** [Makefile:755: device.o] Error 1
make[3]: Leaving directory '/<<PKGBUILDDIR>>/src/powerman'
make[2]: *** [Makefile:407: all-recursive] Error 1
make[2]: Leaving directory '/<<PKGBUILDDIR>>/src'
make[1]: *** [Makefile:469: all-recursive] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
```

This is likely due to Debian (and Ubuntu) [enforcing 64-bit `time_t` format](https://wiki.debian.org/ReleaseGoals/64bit-time) in recent builds after mid-2024. It would be great if you could look into it, and possibly revise the usage of `tv_sec` and `tv_usec`.

--- a/src/powerman/device.c
+++ b/src/powerman/device.c
@@ -1332,8 +1332,8 @@ static bool _process_delay(Device *dev, Action *act, ExecCtx *e,
     /* first time */
     if (!e->processing) {
         if (act->vpf_fun)
-            act->vpf_fun(act->client_id, "delay(%s): %ld.%-6.6ld", dev->name,
-                    delay.tv_sec, delay.tv_usec);
+            act->vpf_fun(act->client_id, "delay(%s): %lld.%-6.6lld", dev->name,
+                    (long long int)delay.tv_sec, (long long int)delay.tv_usec);
         e->processing = true;
         if (gettimeofday(&act->delay_start, NULL) < 0)
             err_exit(true, "gettimeofday");

